### PR TITLE
feat(compiler): Add --no-bulk-memory flag to disable bulk memory ops

### PR DESCRIPTION
**Pull request by phated**
This is a derivative of @ospencer's `near-example` branch that adds a `--no-bulk-memory` CLI flag to turn off our bulk memory operations on platforms that don't support it.

There's still a bug with 2 tests that we need to debug.

### CODE REVIEW

File: compiler/test/runner.re
Comment (phated) [inline compiler/test/runner.re:32]: Just for showing this is working (or not, in the case of the 2 failing tests). Will remove before merging.

File: compiler/src/middle_end/analyze_inline_wasm.re
Comment (phated) [inline compiler/src/middle_end/analyze_inline_wasm.re:253]: I might turn this into a `switch` instead because this is a bit too clever.

**Comment by phated** [comment]
I decided to update the `analyze_inline_wasm.re` file to not use the StringHash maps and instead used switch statements - this allows us to easily vary which ones are enabled using pattern guards!

File: compiler/src/middle_end/analyze_inline_wasm.re
Comment (phated) [inline compiler/src/middle_end/analyze_inline_wasm.re:240]: So much nicer when using a `switch` and pattern guards!

**Comment by phated** [comment]
@grain-lang/core this should be ready to go (tests and all!) 🎉

**Review by peblair — APPROVED** [review]
LGTM, but I will defer to @ospencer on the `analyze_inline_wasm` changes (since I'm less familiar with that part of the code).

**Review by marcusroberts — APPROVED** [review]
Looks good.

File: cli/bin/grain.js
Comment (ospencer) [inline cli/bin/grain.js:116]: What do you think of this description?
```suggestion
  .graincOption("--no-bulk-memory", "polyfill WebAssembly bulk memory instructions")
```

Comment (phated) [reply cli/bin/grain.js:116]: I like it!

**Pull request merged**
